social_core/backends/open_id_connect.py

```diff
@@ -159,6 +159,8 @@ def find_valid_key(self, id_token):
 
         for key in keys:
             if kid is None or kid == key.get('kid'):
+                if 'alg' not in key:
+                    key['alg'] = 'RSA256'
```

Comment:
My identity provider has this issue also. Instead of a fixed default, how about using the first of JWT_ALGORITHMS?

I agree that we don't want to consider the JWT's alg flag since it's so easily [misused](https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries/). 

```suggestion
                    key['alg'] = self.JWT_ALGORITHMS[0]
```